examples/routing/delegated-routing-client/main.go

```diff
@@ -118,6 +102,44 @@ func printIter(w io.Writer, iter iter.ResultIter[types.Record]) error {
 	return nil
 }
 
+func findPeers(w io.Writer, ctx context.Context, client *client.Client, pidStr string) error {
+	// Parses the given Peer ID to lookup the information for.
+	pid, err := peer.Decode(pidStr)
+	if err != nil {
+		return err
+	}
+
+	// Ask for information about the peer with the given peer ID.
+	recordsIter, err := client.FindPeers(ctx, pid)
+	if err != nil {
+		return err
+	}
+	defer recordsIter.Close()
+
+	// The response is streamed. Alternatively, you could use [iter.ReadAll]
+	// to fetch all the results all at once, instead of iterating as they are
+	// streamed.
+	for recordsIter.Next() {
```

Comment:
Can you not reuse `printIter` here since this seems to be a limited copy-paste of that function?

If there are some mechanics here that require this then ok as long as this isn't the kind of thing users are going to regularly stub their toes on.

Reply:
I don't think we can because it's `PeerRecord`, not a generic `Record`. Unless there's some way of weirdly converting that in a single line.

Reply:
Isn't peer record a type of record because `Record` is an interface that `PeerRecord` implements?

Reply:
It is, but Go can't infer that. Maybe you can try, I tried and couldn't manage to make it work. Perhaps too many generics and interfaces together 😃